static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js

```diff
@@ -11,18 +11,49 @@ import { isBlocked } from 'common/modules/commercial/membership-engagement-banne
 import { getSync as getGeoLocation } from 'lib/geolocation';
 import { shouldShowReaderRevenue } from 'common/modules/commercial/contributions-utilities';
 import type { Banner } from 'common/modules/ui/bannerPicker';
+import bean from 'bean';
+import fetchJson from 'lib/fetch-json';
+import reportError from 'lib/report-error';
 
 import {
     submitComponentEvent,
     addTrackingCodesToUrl,
 } from 'common/modules/commercial/acquisitions-ophan';
 import { acquisitionsBannerControlTemplate } from 'common/modules/commercial/templates/acquisitions-banner-control';
+import userPrefs from 'common/modules/user-prefs';
 
-// change messageCode to force redisplay of the message to users who already closed it.
-const messageCode = 'engagement-banner-2018-07-13';
+type BannerDeployLog = {
+    time: string,
+};
+
+const messageCode = 'engagement-banner';
 
-const canDisplayMembershipEngagementBanner = (): Promise<boolean> =>
-    Promise.resolve(shouldShowReaderRevenue());
+const getTimestampOfLastBannerDeploy = (): Promise<string> =>
+    fetchJson('/reader-revenue/contributions-banner-deploy-log', {
```

Comment:
Fetching https://api.nextgen.guardianapps.co.uk/reader-revenue/contributions-banner-deploy-log gives

```javascript
{
  "time": "2018-07-31T11:59:42+0000"
}
```

Before we merge this, we should either set that value to the time we last deployed the banner via the old PR process, or agree with the team that deploying this will cause a sort-of redeploy (at least for people who closed the banner between our last PR-based redeploy and the timestamp above)

Reply:
`userLastClosedBannerAt` will check for a non-existent string in local storage to convert to a Date which results in `Thu Jan 01 1970 01:00:00 GMT+0100`. If we don't want to display the banner to everyone when this is merged, we need to change the json timestamp to be the same value. (Only changing the banner timestamp means we either show everyone or no one, we cannot only show it to those who closed the banner between last PR-based redeploy).